### PR TITLE
Sa1205 fix

### LINKED ISSUE #937: SA1205 (PartialElementsMustDeclareAccess) code fix inserts modifier at the wrong location
The code fix is placingthe modifier before xml documentation.

Im fixing that

### PR DESCRIPTION
**Pull request by pdelvo**
Fixes #937.

SA1400 already had a method which did the right thing so I just put it into a helper class. I also added a regression test.

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers/Helpers/TypeDeclarationSyntaxHelper.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
+﻿namespace StyleCop.Analyzers.Helpers
+{
+    using System.Linq;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+
+    internal class TypeDeclarationSyntaxHelper
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/Helpers/TypeDeclarationSyntaxHelper.cs]: :bulb: `static class`

:question: SA1400 uses this for updating modifiers of both types and members. Should it have a different name?

Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers/Helpers/TypeDeclarationSyntaxHelper.cs]: You're right. Do you have a suggestion?

Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/Helpers/TypeDeclarationSyntaxHelper.cs]: `DeclarationModifiersHelper`?

Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers/Helpers/TypeDeclarationSyntaxHelper.cs]: fixed

**Comment by pdelvo** [comment]
I implemented the PR feedback

**Pull request merged**